include/circt/Dialect/FIRRTL/Types.td

```diff
@@ -19,6 +19,14 @@ def UInt1Type : Type<CPred<"$_self.isa<UIntType>() && "
                            "$_self.cast<UIntType>().getWidth() == 1">,
                            "UInt<1>">,
                 BuildableType<"UIntType::get($_builder.getContext(), 1)">;
+def ResetType : Type<Or<[
+                       CPred<"$_self.isa<ResetType>()">,
```

Comment:
Random though - you could split this out into a `isValidResetType(Type x)` sort of helper function or an `FIRRTLType::isValidResetType()` predicate instead of encoding this into ODS.  It is probably a generally useful predicate to have.

Reply:
Fixed in the rebase. I added a `FIRRTLType:isResetType` (following the naming convention of `FIRRTLType::isPassiveType`). Yes, this does seem generally useful. Good idea.